com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -125,47 +161,34 @@ internal NetworkSerializer __beginSendClientRpc(uint rpcMethodId, ClientRpcParam
             if (IsHost && containsServerClientId)
             {
                 //Always write to the next frame's inbound queue
-                writer = rpcQueueContainer.BeginAddQueueItemToFrame(RpcQueueContainer.QueueItemType.ClientRpc, Time.realtimeSinceStartup, transportChannel,
-                    NetworkManager.ServerClientId, null, RpcQueueHistoryFrame.QueueFrameType.Inbound, clientRpcParams.Send.UpdateStage);
+                writer = messageQueueContainer.BeginAddQueueItemToFrame(MessageQueueContainer.MessageType.ClientRpc, Time.realtimeSinceStartup, transportChannel,
+                    NetworkManager.ServerClientId, null, MessageQueueHistoryFrame.QueueFrameType.Inbound, clientRpcParams.Send.UpdateStage);
 
                 //Handle sending to the other clients, if so the above notes explain why this code is here (a temporary patch-fix)
                 if (clientIds.Length > 1)
                 {
                     //Set the loopback frame
-                    rpcQueueContainer.SetLoopBackFrameItem(clientRpcParams.Send.UpdateStage);
+                    messageQueueContainer.SetLoopBackFrameItem(clientRpcParams.Send.UpdateStage);
 
                     //Switch to the outbound queue
-                    writer = rpcQueueContainer.BeginAddQueueItemToFrame(RpcQueueContainer.QueueItemType.ClientRpc, Time.realtimeSinceStartup, transportChannel, NetworkObjectId,
-                        clientIds, RpcQueueHistoryFrame.QueueFrameType.Outbound, NetworkUpdateStage.PostLateUpdate);
-
-                    if (!isUsingBatching)
-                    {
-                        writer.WriteByte(NetworkConstants.CLIENT_RPC); // MessageType
-                    }
-                }
-                else
-                {
-                    if (!isUsingBatching)
-                    {
-                        writer.WriteByte(NetworkConstants.CLIENT_RPC); // MessageType
-                    }
+                    writer = messageQueueContainer.BeginAddQueueItemToFrame(MessageQueueContainer.MessageType.ClientRpc, Time.realtimeSinceStartup, transportChannel, NetworkObjectId,
+                        clientIds, MessageQueueHistoryFrame.QueueFrameType.Outbound, NetworkUpdateStage.PostLateUpdate);
+                    writer.WriteByte((byte)MessageQueueContainer.MessageType.ClientRpc);
+                    writer.WriteByte((byte)clientRpcParams.Send.UpdateStage); // NetworkUpdateStage
```

Comment:
This bit and lines 182-185 - please wrap them in a function, they seem to be identical.

Reply:
I'm curious if there's a team guideline on when something is large enough to functionalize... To me, functionalizing three lines used twice is a little overkill and leads to more function call overhead and reduced readability of the code.

In this case I think there's a way to pull these lines out of the if/else and unify them without a function so I'll do that, but I think it's possible to over-functionalize just as much as it is to under-functionalize.

Reply:
Done.